App/Views/DealsPage.xaml

```diff
@@ -49,30 +49,66 @@
                 <Grid RowDefinitions="220,50"
                       x:Name="DealBlock"
                       Padding="10">
-                    <Grid.Behaviors>
-                        <mct:TouchBehavior Command="{Binding Navigate}"
-                                           BindingContext="{Binding BindingContext, 
-                                                                    Source={x:Reference DealBlock},
-                                                                    x:DataType=Grid}"/>
-                    </Grid.Behaviors>
                     <Border Grid.Row="0"
-                            StrokeShape="RoundRectangle 8">
+                            StrokeShape="RoundRectangle 8"
+                            x:Name="thumbnail">
                         <Image Source="{Binding Image}"
                                HorizontalOptions="Fill"
                                VerticalOptions="Fill"
                                Aspect="Fill"/>
+
+                        <Border.Behaviors>
+                            <mct:TouchBehavior Command="{Binding Navigate}"
+                                           BindingContext="{Binding BindingContext, 
+                                                                    Source={x:Reference thumbnail},
+                                                                    x:DataType=Border}"/>
+                        </Border.Behaviors>
                     </Border>
-                    <Border Grid.Row="0"
-                            StrokeShape="RoundRectangle 0,8,8,0"
-                            HorizontalOptions="End"
-                            VerticalOptions="Start"
-                            HeightRequest="50"
-                            WidthRequest="100"
-                            BackgroundColor="White">
-                        <Image Source="{Binding Partner.Logo}"
-                               HorizontalOptions="Fill"
-                               VerticalOptions="Fill"/>
-                    </Border>
+                    <Grid Grid.Row="0"
+                          RowDefinitions="50,*">
+                        <Border Grid.Row="0"
+                                StrokeShape="RoundRectangle 0,8,20,0"
+                                HorizontalOptions="End"
+                                VerticalOptions="Start"
+                                HeightRequest="50"
+                                WidthRequest="100"
+                                x:Name="partner"
+                                BackgroundColor="White">
+                            <Image Source="{Binding Partner.Logo}"
+                                   HorizontalOptions="Fill"
+                                   VerticalOptions="Fill"/>
+
+                            <Border.Behaviors>
+                                <mct:TouchBehavior Command="{Binding Navigate}"
+                                           BindingContext="{Binding BindingContext, 
```

Comment:
I know it looks a bit odd to repeat the navigation command everywhere but there is currently no other way to prevent that command to overshadow the share button and be triggered no matter where we press